src/graph/transform/to_simple.cc

```diff
@@ -0,0 +1,100 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file graph/transform/to_simple.cc
+ * \brief Convert multigraphs to simple graphs
+ */
+
+#include <dgl/base_heterograph.h>
+#include <dgl/transform.h>
+#include <dgl/array.h>
+#include <dgl/packed_func_ext.h>
+#include <vector>
+#include <utility>
+#include "../unit_graph.h"
+#include "../../c_api_common.h"
+
+namespace dgl {
+
+using namespace dgl::runtime;
+using namespace dgl::aten;
+
+namespace transform {
+
+std::tuple<HeteroGraphPtr, std::vector<IdArray>, std::vector<IdArray>>
+ToSimpleGraph(const HeteroGraphPtr graph) {
+  const int64_t num_etypes = graph->NumEdgeTypes();
+  const auto metagraph = graph->meta_graph();
+
+  std::vector<IdArray> counts(num_etypes), edge_maps(num_etypes);
+  std::vector<HeteroGraphPtr> rel_graphs(num_etypes);
+
+  for (int64_t etype = 0; etype < num_etypes; ++etype) {
+    const auto vtypes = graph->GetEndpointTypes(etype);
+    const COOMatrix adj = graph->GetCOOMatrix(etype);
+    const COOMatrix sorted_adj = COOSort(adj, true);
+    const IdArray eids_shuffled = sorted_adj.data;
+    const COOMatrix coalesced_adj = COOCoalesce(sorted_adj);
```

Comment:
It might worth implementing `Coalesce` for CSR too.

Reply:
Can we do this after someone complains about the speed of `to_simple` on big graphs?